openfisca_france/scenarios.py

```diff
@@ -61,631 +60,386 @@ def init_single_entity(self, axes = None, enfants = None, famille = None, foyer_
-                            ),
-                        conv.empty_to_none,
-                        conv.not_none,
-                        ),
-                    ),
-                default = conv.noop,
-                )(test_case, state = state)
+                                menage[u'conjoint'] = individu_id
+                                individus_without_menage.remove(individu_id)
+                            elif other_role == u'conjoint' and menage[u'personne_de_reference'] is None:
+                                # Quand l'individu n'est pas encore dans un ménage, mais qu'il est déclarant
+                                # dans une foyer fiscal, qu'il y a un autre déclarant dans ce foyer fiscal et
+                                # que cet autre déclarant est conjoint dans un ménage et qu'il n'y a pas de
+                                # personne de référence dans ce ménage, alors ajoute l'individu comme personne
+                                # de référence de ce ménage.
+                                menage[u'personne_de_reference'] = individu_id
+                                individus_without_menage.remove(individu_id)
+                            break
+                elif foyer_fiscal_role == u'personnes_a_charge' and foyer_fiscal[u'declarants']:
+                    for declarant_id in foyer_fiscal[u'declarants']:
+                        menage, other_role = find_menage_and_role(test_case, declarant_id)
+                        if other_role in (u'personne_de_reference', u'conjoint'):
+                            # Quand l'individu n'est pas encore dans un ménage, mais qu'il est personne à charge
+                            # dans un foyer fiscal, qu'il y a un déclarant dans ce foyer fiscal et que ce
+                            # déclarant est personne de référence ou conjoint dans un ménage, alors ajoute
+                            # l'individu comme enfant de ce ménage.
+                            menage[u'enfants'].append(individu_id)
+                            individus_without_menage.remove(individu_id)
+                            break
 
-            return test_case, error
+            if individu_id in individus_without_menage:
+                # L'individu n'est toujours pas affecté à un ménage.
+                if new_menage[u'personne_de_reference'] is None:
+                    new_menage[u'personne_de_reference'] = individu_id
+                elif new_menage[u'conjoint'] is None:
+                    new_menage[u'conjoint'] = individu_id
+                else:
+                    new_menage[u'enfants'].append(individu_id)
+                if new_menage_id is None:
+                    new_menage[u'id'] = new_menage_id = unicode(uuid.uuid4())
+                    test_case[u'menages'].append(new_menage)
+                individus_without_menage.remove(individu_id)
+
+        return test_case
 
-        return json_or_python_to_test_case
 
     def suggest(self):
```

Comment:
Yes, `suggest` was introduced for the UI and could be removed (this would imply to remove [a flag](https://github.com/openfisca/openfisca-web-api/blob/d790243123c72fd198cc29a63d6dd4dedb316510/openfisca_web_api/controllers/simulate.py#L174) in the API code).

> why it was separated from the rest of the test_case processing ?

For code clarity I think.